controller/scheduler/swarm.py

```diff
@@ -0,0 +1,147 @@
+import re
+import time
+from docker import Client
+from django.conf import settings
+from .states import JobState
+
+MATCH = re.compile(
+    '(?P<app>[a-z0-9-]+)_?(?P<version>v[0-9]+)?\.?(?P<c_type>[a-z-_]+)?.(?P<c_num>[0-9]+)')
+
+
+class SwarmClient(object):
+    def __init__(self, target, auth, options, pkey):
+        self.target = settings.SWARM_HOST
+        # single global connection
+        self.registry = settings.REGISTRY_HOST+":"+settings.REGISTRY_PORT
+        self.docker_cli = Client(base_url='tcp://'+self.target+':'+"2395",
+                                 timeout=1200, version='1.17')
+
+    def create(self, name, image, command='', template=None, **kwargs):
+        """Create a container"""
+        cimage = self.registry+"/"+image
+        cname = name
+        ccommand = command
+        affinity = "affinity:container!=~/"+(re.split("_v\d.", cname)[0]) + "*/"
+        l = locals().copy()
+        l.update(re.match(MATCH, name).groupdict())
+        mem = kwargs.get('memory', {}).get(l['c_type'], None)
+        if mem:
+            mem = mem.lower()
+            if mem[-2:-1].isalpha() and mem[-1].isalpha():
+                mem = mem[:-1]
+        cpu = kwargs.get('cpu', {}).get(l['c_type'], None)
+        self.docker_cli.create_container(image=cimage, name=cname,
+                                         command=ccommand.encode('utf-8'), mem_limit=mem,
+                                         cpu_shares=cpu,
+                                         environment=[affinity])
+        self.docker_cli.stop(name)
+
+    def start(self, name):
+        """
+        Start a container
+        """
+        self.docker_cli.start(name, publish_all_ports=True)
+
+        return
```

Comment:
Empty `return` statements aren't necessary in python methods--let's remove all these.